src/DataTransferObject.php

```diff
@@ -17,14 +17,27 @@ abstract class DataTransferObject
 
     protected array $onlyKeys = [];
 
-    public function __construct(...$args)
+    public static function new(...$args): static
+    {
+        $dataTransferObject = new static();
+
+        $class = new DataTransferObjectClass($dataTransferObject);
+
+        return $dataTransferObject
+            ->setUp($class, ...$args)
+            ->validate($class);
+    }
+
+    private function __construct()
+    {
+    }
+
+    protected function setUp(DataTransferObjectClass $class, ...$args): static
```

Comment:
This could potentially be:

```php
public static function newWithoutValidation(...$arguments): static
{
    //
}
```

Reply:
Absolutely, feel free to refactor if you're working in the code.

Reply:
> Absolutely, feel free to refactor if you're working in the code.

Will do!